internal/gather/gather_bytes_test.go

```diff
@@ -173,6 +182,157 @@ func TestGatherBytesReadSeeker(t *testing.T) {
 	require.Error(t, err)
 }
 
+func TestGatherBytesReaderAtErrorResponses(t *testing.T) {
+	// 3.7 times the internal chunk size
+	contentBuf := make([]byte, int(float64(defaultAllocator.chunkSize)*3.7))
+	for i := range contentBuf {
+		contentBuf[i] = uint8(i % math.MaxUint8)
+	}
+
+	tcs := []struct {
+		inBsLen   int
+		inOff     int64
+		expectErr error
+		expectN   int
+	}{
+		{
+			inBsLen:   1 << 10,
+			inOff:     -1,
+			expectErr: ErrInvalidOffset,
+			expectN:   0,
+		},
+		{
+			inBsLen:   1 << 10,
+			inOff:     math.MaxInt64,
+			expectErr: io.EOF,
+			expectN:   0,
+		},
+		{
+			inBsLen:   0,
+			inOff:     -1,
+			expectErr: ErrInvalidOffset,
+			expectN:   0,
+		},
+		{
+			inBsLen: 0,
+			inOff:   math.MaxInt64,
+			expectN: 0,
+		},
+	}
+	for i, tc := range tcs {
+		t.Run(fmt.Sprintf("%d: %d %d %d", i, tc.inBsLen, tc.inOff, tc.expectN), func(t *testing.T) {
+			// tmp is an empty buffer that will supply some bytes
+			// for testing
+			var wrt WriteBuffer
+			defer wrt.Close()
+
+			wrt.Append(contentBuf)
+			require.Equalf(t, defaultAllocator.chunkSize, wrt.alloc.chunkSize,
+				"this test expects that the default-allocator will be used, but we are using: %#v", wrt.alloc)
+
+			// get the reader out of the WriteBuffer so we can read what was written
+			// (presume all 0s)
+			reader := wrt.inner.Reader()
+			defer reader.Close() //nolint:errcheck
+
+			// get the reader as a ReaderAt
+			readerAt := reader.(io.ReaderAt)
+
+			// make an output buffer of the required length
+			bs := make([]byte, tc.inBsLen)
+
+			n, err := readerAt.ReadAt(bs, tc.inOff)
+			require.ErrorIs(t, err, tc.expectErr)
+			require.Equal(t, tc.expectN, n)
+		})
+	}
+}
+
+func TestGatherBytesReaderAtVariableInputBufferSizes(t *testing.T) {
+	const inputBufferMaxMultiplier = 4.0 // maximum number of times the internal chunk size
+
+	contentBuf := make([]byte, defaultAllocator.chunkSize*inputBufferMaxMultiplier)
+	for i := range contentBuf {
+		contentBuf[i] = uint8(i % math.MaxUint8)
+	}
+
+	type testCase struct {
+		name            string
+		inputBufferSize int
+	}
+	// Test some interesting input buffer sizes from a minimum three
+	// times the size of the internal allocator byte slices.
```

Comment:
Comment can be updated I think
```suggestion
	// Test some interesting input buffer sizes from a 1-byte buffer to many multiples 
	// of the internal allocator chunk size.
```